src/views/DatasetManager/components/CreateDatasetButton/components/DatasetCreationParameters/DatasetCreationParametersHook.js

```diff
@@ -0,0 +1,188 @@
+        parameters: [
+          {
+            id: `location`,
+            varType: 'string',
+            labels: { en: 'Path', fr: 'Chemin' },
+          },
+        ],
+      },
+      {
+        id: DATASET_SOURCE_TYPE.LOCAL_FILE,
+        labels: {
+          en: 'Graph Format from Local File',
+          fr: 'Format Graph depuis un fichier local',
+        },
+        parameters: [{ id: `file`, varType: '%DATASETID%', labels: { en: '', fr: '' } }],
+      },
+      {
+        id: DATASET_SOURCE_TYPE.NONE,
+        labels: {
+          en: 'Empty',
+          fr: 'Dataset vide',
+        },
+        parameters: [],
+      },
+    ];
+
+    dataSources.forEach((dataSource) => {
+      dataSource.parameters.forEach((parameter) => (parameter.id = `${dataSource.id}.${parameter.id}`));
+    });
+
+    TranslationUtils.addTranslationRunTemplateLabels(dataSources);
+    TranslationUtils.addTranslationParametersLabels(dataSources.flatMap((dataSource) => dataSource?.parameters));
+
+    return dataSources;
+  }, []);
+
+  const dataSourceRunTemplates = useMemo(() => {
+    const dataSources = solutionData.runTemplates.filter((runTemplate) => runTemplate?.tags.includes('datasource'));
+
+    const parameters = solutionData.parameters;
+    const runTemplatesParameters = solutionData.runTemplatesParametersIdsDict;
+
+    const dataSourcesWithParameters = dataSources.map((dataSource) => {
+      const dataSourceWithParameters = { ...dataSource };
+      dataSourceWithParameters.parameters = parameters.filter((parameter) =>
+        runTemplatesParameters[dataSource.id].includes(parameter.id)
+      );
+      return dataSourceWithParameters;
+    });
+
+    return [...hardCodedDataSources, ...dataSourcesWithParameters];
+  }, [
+    solutionData.parameters,
+    solutionData.runTemplates,
+    solutionData.runTemplatesParametersIdsDict,
+    hardCodedDataSources,
+  ]);
+
+  const dataSourceTypeEnumValues = useMemo(() => {
+    return [
+      ...dataSourceRunTemplates.map((dataSource) => {
+        return {
+          key: dataSource.id,
+          value: t(TranslationUtils.getRunTemplateTranslationKey(dataSource.id), dataSource.label),
+        };
+      }),
+    ];
+  }, [t, dataSourceRunTemplates]);
+
+  const getDataSource = useCallback(
+    (dataSourceId) => {
+      return dataSourceRunTemplates.find((dataSource) => dataSource.id === dataSourceId);
+    },
+    [dataSourceRunTemplates]
+  );
+
+  const isDataSourceTypeRunner = useCallback(
+    (dataSourceId) => {
+      return getDataSource(dataSourceId)?.tags != null && getDataSource(dataSourceId)?.tags?.includes('datasource');
+    },
+    [getDataSource]
+  );
+
+  const getParameter = useCallback(
+    (parameterId) => {
+      for (const dataSource of dataSourceRunTemplates) {
+        const parameter = dataSource?.parameters.find((parameter) => parameter.id === parameterId);
+        if (parameter != null) return parameter;
+      }
+    },
+    [dataSourceRunTemplates]
+  );
+
```

Comment:
Can't we use a simple `find` in the solution parameters instead?

Reply:
I think my algorithm is more accurate and so more efficient. It executes a search within _sourceRuntemplates_ already filterded as _datasource_. So, the `find` function executes very smaller loops than a "global" find in whole solution including all parameters of the solution.